xds/src/main/java/io/grpc/xds/RbacFilter.java

```diff
@@ -0,0 +1,325 @@
+    }
+    Any anyMessage = (Any) rawProtoMessage;
+    try {
+      rbacProto = anyMessage.unpack(RBAC.class);
+    } catch (InvalidProtocolBufferException e) {
+      return ConfigOrError.fromError("Invalid proto: " + e);
+    }
+    return parseRbacConfig(rbacProto);
+  }
+
+  @VisibleForTesting
+  static ConfigOrError<RbacConfig> parseRbacConfig(RBAC rbac) {
+    checkNotNull(rbac, "rbac");
+    if (!rbac.hasRules()) {
+      return null;
+    }
+    io.envoyproxy.envoy.config.rbac.v3.RBAC rbacConfig = rbac.getRules();
+    GrpcAuthorizationEngine.Action authAction;
+    switch (rbacConfig.getAction()) {
+      case ALLOW:
+        authAction = GrpcAuthorizationEngine.Action.ALLOW;
+        break;
+      case DENY:
+        authAction = GrpcAuthorizationEngine.Action.DENY;
+        break;
+      case LOG:
+      default:
+        return null;
+    }
+    Map<String, Policy> policyMap = rbacConfig.getPoliciesMap();
+    List<GrpcAuthorizationEngine.PolicyMatcher> policyMatchers = new ArrayList<>();
+    for (Map.Entry<String, Policy> entry: policyMap.entrySet()) {
+      try {
+        Policy policy = entry.getValue();
+        if (policy.hasCondition() || policy.hasCheckedCondition()) {
+          return ConfigOrError.fromError(
+                  "Policy.condition and Policy.checked_condition must not set: " + entry.getKey());
+        }
+        policyMatchers.add(new PolicyMatcher(entry.getKey(),
+                parsePermissionList(policy.getPermissionsList()),
+                parsePrincipalList(policy.getPrincipalsList())));
+      } catch (Exception e) {
+        return ConfigOrError.fromError("Encountered error parsing policy: " + e);
+      }
+    }
+    return ConfigOrError.fromConfig(RbacConfig.create(new AuthConfig(policyMatchers, authAction)));
+  }
+
+  @Override
+  public ConfigOrError<RbacConfig> parseFilterConfigOverride(Message rawProtoMessage) {
+    RBACPerRoute rbacPerRoute;
+    if (!(rawProtoMessage instanceof Any)) {
+      return ConfigOrError.fromError("Invalid config type: " + rawProtoMessage.getClass());
+    }
+    Any anyMessage = (Any) rawProtoMessage;
+    try {
+      rbacPerRoute = anyMessage.unpack(RBACPerRoute.class);
+    } catch (InvalidProtocolBufferException e) {
+      return ConfigOrError.fromError("Invalid proto: " + e);
+    }
+    if (rbacPerRoute.hasRbac()) {
+      return parseRbacConfig(rbacPerRoute.getRbac());
+    } else {
+      return ConfigOrError.fromConfig(RbacConfig.create(null));
+    }
+  }
+
+  @Override
+  public ServerInterceptor buildServerInterceptor(FilterConfig config,
+                                                  @Nullable FilterConfig overrideConfig) {
+    checkNotNull(config, "config");
+    if (overrideConfig != null) {
+      config = overrideConfig;
+    }
+    AuthConfig authConfig = ((RbacConfig) config).authConfig();
+    return authConfig == null ? null : generateAuthorizationInterceptor(authConfig);
```

Comment:
Declare in `Filter. ServerInterceptorBuilder ` to return `@Nullable` or return no-op interceptor here.

Reply:
Let's do `Nullable`, this is also what FaultFilter is doing.